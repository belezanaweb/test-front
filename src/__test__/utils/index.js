import checkPropTypes from 'check-prop-types'

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`)
  return wrapper
}

export const checkProps = (component, expectedProps) => {
  const propsError = checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
  return propsError
}
