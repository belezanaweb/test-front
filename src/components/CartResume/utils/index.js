export const normalizeCartData = data => {
  return {
    productsQuantity: data.items.length,
    subTotal: data.subTotal.toFixed(2).replace('.', ','),
    shipping: data.shippingTotal.toFixed(2).replace('.', ','),
    discount: data.discount.toFixed(2).replace('.', ','),
    total: data.total.toFixed(2).replace('.', ','),
  };
};
