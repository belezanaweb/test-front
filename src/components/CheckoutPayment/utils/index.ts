import * as Yup from 'yup'

const errors = {
  name: 'Insira um nome válido',
  code: 'Insira um código válido',
  number: 'Insira um número válido',
  expiration: 'Insira uma data válida',
}

export const initialValues = {
  creditCardName: '',
  creditCardNumber: '',
  creditCardExpiration: '',
  creditCardCode: '',
}

const regexOnlyNumbers = /[^\d.-]+/g

export const CheckoutPaymentSchema = Yup.object().shape({
  creditCardName: Yup.string()
    .min(2, errors.name)
    .matches(/^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi, errors.name)
    .required(errors.name),
  creditCardNumber: Yup.string()
    .min(2, errors.number)
    .test('len', errors.number, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 16
    })
    .required(errors.number),
  creditCardExpiration: Yup.string()
    .min(2, errors.name)
    .test('len', errors.expiration, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 4
    })
    .required(errors.expiration),
  creditCardCode: Yup.string()
    .matches(/^[0-9]+$/, errors.code)
    .test('len', errors.code, (val) => val.length === 3)
    .required(errors.code),
})
