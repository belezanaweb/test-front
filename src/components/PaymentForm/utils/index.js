export const formatCardNumber = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{4})(?=\d)/g, '$1 ')
    .substr(0, 19);
};

export const formatHolderName = value => {
  return value.replace(/[^a-zA-Z\u00C0-\u017F\s]/gu, '').substr(0, 60);
};

export const formatExpirationDate = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d{0,4})/, '$1/$2')
    .substr(0, 7);
};

export const formatCVVCode = value => {
  return value.replace(/\D/g, '').substr(0, 3);
};

const validateCreditCard = cardNumber => {
  cardNumber = cardNumber.replace(/\s/g, '');

  if (cardNumber.length !== 16) {
    return false;
  }

  const digits = cardNumber.split('').map(Number);

  // Algoritmo de Luhn para validar o número do cartão
  let sum = 0;

  for (let i = 0; i < digits.length; i++) {
    let digit = digits[i];

    if (i % 2 === 0) {
      digit *= 2;

      if (digit > 9) {
        digit -= 9;
      }
    }

    sum += digit;
  }

  // O número do cartão é válido se a soma dos dígitos for um múltiplo de 10
  return sum % 10 === 0;
};

const isInvalidDate = dateString => {
  const today = new Date();
  const [month, year] = dateString.split('/');
  if (month === '00') {
    return true;
  }
  const userDate = new Date(Number(year), Number(month) - 1);

  return userDate < new Date(today.getFullYear(), today.getMonth());
};

export const validateForm = state => {
  const errors = {};

  if (!validateCreditCard(state.cardNumber)) {
    errors.cardNumber = 'insira um número de cartão válido';
  }

  if (!state.cardHolderName) {
    errors.cardHolderName = 'insira um nome válido';
  }

  if (state.expirationDate.length < 7 || isInvalidDate(state.expirationDate)) {
    errors.expirationDate = 'insira uma data válida';
  }

  if (state.cvv.length < 3) {
    errors.cvv = 'insira um cvv válido';
  }

  return errors;
};
