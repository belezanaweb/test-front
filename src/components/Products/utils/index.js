export const normalizeCartData = data => {
  return data.items.reduce((acc, curr) => {
    const formattedProduct = {
      imageUrl: curr.product.imageObjects[0].small,
      productName: curr.product.name,
      price: curr.product.priceSpecification.price,
      maxPrice: curr.product.priceSpecification.maxPrice,
    };

    return [...acc, formattedProduct];
  }, []);
};
