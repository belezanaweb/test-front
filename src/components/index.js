import Button from './Button'
import Header from './Header'
import Section from './Section'
import CartList from './CartList'
import CartSummary from './CartSummary'

export { Button, Header, Section, CartList, CartSummary }
