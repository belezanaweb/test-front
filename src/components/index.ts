export { default as Checkout } from './Checkout'
export { default as Tabs } from './Tabs'
export { default as Card } from './Card'
export { default as SkeletonCheckout } from './SkeletonCheckout'
export { default as Skeleton } from './Skeleton'
export { default as Text } from './Text'
export { default as CheckoutItemsSummary } from './CheckouItemsSummary'
export { default as Button } from './Button'
export { default as CheckoutPayment } from './CheckoutPayment'
export { default as Input } from './Input'
