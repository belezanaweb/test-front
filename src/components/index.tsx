export { default as AppBar } from "./AppBar";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as Input } from "./Input";
export { default as Loading } from "./Loading";
export { default as ProductItem } from "./ProductItem";
export { default as Summary } from "./Summary";
export { default as SummaryItem } from "./SummaryItem";
export { default as TabItem } from "./TabItem";
export { default as Tabs } from "./Tabs";
export { default as Theme } from "./Theme";
export { default as Typography } from "./Typography";
