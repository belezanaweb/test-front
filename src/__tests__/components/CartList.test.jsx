import React from 'react'
import { render } from '../../utils/test-utils'
import { CartList } from '../../components'

const makeSut = (props) => {
  const mockCartItems = [
    {
      product: {
        sku: 1,
        name: 'Sample product 1',
        imageObjects: [{ small: 'teste' }],
        priceSpecification: {
          price: 100.0
        }
      }
    },
    {
      product: {
        sku: 2,
        name: 'Sample product 2',
        priceSpecification: {
          price: 100.0
        }
      }
    }
  ]

  return render(<CartList cartItems={mockCartItems} {...props} />)
}

describe('CartList', () => {
  test('should render without errors', () => {
    const { container } = render(<CartList />)
    expect(container).toBeInTheDocument()
  })

  test('should component render items', () => {
    const { container } = makeSut()

    const listElement = container.querySelector('ul')

    expect(listElement).toBeInTheDocument()
    expect(listElement?.children).toHaveLength(2)
  })

  test('should match with snapshot', () => {
    const { container } = makeSut()
    expect(container).toMatchSnapshot()
  })
})
