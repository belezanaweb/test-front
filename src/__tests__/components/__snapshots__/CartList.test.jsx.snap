// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList should match with snapshot 1`] = `
<div>
  <div
    class="sc-fujyAs bSHCym"
  >
    <ul>
      <li
        class="sc-pNWdM eNrdGQ"
      >
        <img
          class="sc-jrsJWt dCtIRT"
          src="teste"
        />
        <div
          class="sc-kEqXSa bzaWxE"
        >
          <p
            class="sc-iqAclL vozpB"
          >
            Sample product 1
          </p>
          <span
            class="sc-crzoAE gpMDvT"
          >
            R$ 100,00
          </span>
        </div>
      </li>
      <li
        class="sc-pNWdM eNrdGQ"
      >
        <img
          class="sc-jrsJWt dCtIRT"
        />
        <div
          class="sc-kEqXSa bzaWxE"
        >
          <p
            class="sc-iqAclL vozpB"
          >
            Sample product 2
          </p>
          <span
            class="sc-crzoAE gpMDvT"
          >
            R$ 100,00
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
